packages/core-contracts/src/contracts/FleetProxy.sol

```diff
@@ -159,11 +159,23 @@ contract FleetProxy is
         // 5. Approve the bridge router to transfer the assets
         IERC20(asset).forceApprove(address(bridgeRouter), params.amount);
 
-        // 6. Get source chain ark address from registry - reverts if not found
-        address arkAddress = _getSourceChainArk(hubChainId);
-
-        // 7. Use BridgeQueue to queue a transfer of assets back to source chain's CrossChainArk
-        bridgeRouter.executeTransferAssets(params);
+        // 6. Get source-chain Ark address from the registry – reverts if not found
+        address arkAddress = _getSourceChainArk(params.destinationChainId);
+
+        // 7. Build a new ExecuteTransferParams struct with the correct Ark recipient
```

Comment:
why is this required if the address is verified here: https://github.com/OasisDEX/summer-earn-protocol/blob/0440c078102e2bcc789ba9a4e92659fff819afac/packages/core-contracts/src/contracts/FleetProxy.sol#L141

```ts
        if (params.recipient != _getSourceChainArk(params.destinationChainId))
            revert InvalidRecipient();
```